sys/linux/net_80211.txt

```diff
@@ -55,3 +60,201 @@ define JOIN_IBSS_BG_SCAN	0x1
 define JOIN_IBSS_BG_NO_SCAN	0x2
 
 join_ibss_modes = JOIN_IBSS_NO_SCAN, JOIN_IBSS_BG_SCAN, JOIN_IBSS_BG_NO_SCAN
+
+################################################################################
+#  Common structs and enums
+################################################################################
+
+# As defined in drivers/net/wireless/mac80211_hwsim.c
+ieee80211_channel_freq_mhz = 2412, 2417, 2422, 2427, 2432, 2437, 2442, 2447, 2452, 2457, 2462, 2467, 2472, 2484, 5180, 5200, 5220, 5240, 5260, 5280, 5300, 5320, 5500, 5520, 5540, 5560, 5580, 5600, 5620, 5640, 5660, 5680, 5700, 5745, 5765, 5785, 5805, 5825, 5845, 5855, 5860, 5865, 5870, 5875, 5880, 5885, 5890, 5895, 5900, 5905, 5910, 5915, 5920, 5925
+
+ieee80211_raw_rates = 10, 20, 55, 60, 90, 110, 120, 180, 240, 360, 480, 540
+
+# Combined set of 802.11 rates for 5Mhz, 10Mhz and other channel widths
+# Only rates for drivers/net/wireless/mac80211_hwsim.c are defined
+ieee80211_rate_label = 1, 2, 3, 4, 5, 6, 9, 11, 12, 18, 22, 24, 27, 36, 48, 54, 72, 96, 108
+
+# Rates as they are represented (see 9.4.2.3 of IEEEE 802.11-2016)
+ieee80211_rate {
+	label		flags[ieee80211_rate_label, int8:7]
+	mandatory	int8:1
+} [packed]
+
+type ieee80211_beacon_interval[BASE_TYPE] [
+	default	const[100, BASE_TYPE]
+	random	BASE_TYPE
+]
+
+type ieee80211_timestamp int64
+
+# Pseudo syscalls and initially created devices use the default frequency below 
+type ieee80211_frequency_mhz[BASE_TYPE] [
+	default	const[2412, BASE_TYPE]
+	random	flags[ieee80211_channel_freq_mhz, BASE_TYPE]
+]
+
+# Information Element structure (see 9.4.2.1 of IEEE 802.11-2016)
```

Comment:
Could you add a direct link to this document? Do other structs in this file also come from it? If not, we need links to all the other documents too.

Reply:
Added a link. This is the only document that is required.

Reply:
There's no direct link, right?

Reply:
Yes, there's no direct link at the official sources.